pkg/config/config/config.go

```diff
@@ -663,6 +664,27 @@ func UseViper(v *viper.Viper) error {
 		}
 	}
 
+	cspAllowList := map[string]string{}
+	cspPerContext := map[string]map[string]string{}
+	cspList := v.GetStringMap("csp_allowlist")
+	for key, value := range cspList {
+		if val, ok := value.(string); ok {
+			cspAllowList[key] = val
+		} else if val, ok := value.(map[string]interface{}); ok && key == "contexts" {
```

Comment:
probably you should first check for `key == "contexts"` and only try to decode if it maches.
This would avoid trying to decode something not intended to be decoded here.

Reply:
I don't understand what you want to do. Is it an optimization for returning faster an error if the config is invalid?

Reply:
finally no need to change anything here